### PR TITLE
Textfiles code compliance improvements

### PR DESCRIPTION
**Pull request by MaanooAk**
### Code compliance improvements:
- Print line number of an issue
- Added missing space issue (detects `){` and `if/for/while(`)
- Added extra space issue (detects spaces before `;`)
- Can be extended

> For the current master the `make checkfast` print:
>
> screenshot from 2017-01-04 17-06-26

### How it works:
First the whole file string is searched and if at least an issue is found somewhere, the file is searched line by line (in order to generate proper description with  the line and line number).
Regular expressions and generators are used when possible.

### CODE REVIEW

**Comment by MaanooAk** [comment]
The tests are failing because of the new issue detection. Before merge we should fix the warning (if this pr is in the right direction)

**Comment by TheJJ** [comment]
You are: awesome.
This is very much in the right direction. The thing we didn't think about back then was adding exceptions for Cython generated files.

However: I would put the `.cpp` and `.h` checks in a separate file like `cppstyle.py`. And if the `textfiles.py` then needs to share code with the `cppstyle.py`, use the common functions by importing `textfiles` from `cppstyle.py`

Thanks :+1:

**Comment by MaanooAk** [comment]
The `textfiles.py` is not compatible with the `only_changed_files`. Is there a reason or should I change it?

File: buildsystem/codecompliance/__main__.py [hunk 1]
```diff
@@ -166,6 +169,10 @@ def find_all_issues(args, check_files=None):
         from .pystyle import find_issues
         yield from find_issues(check_files, ('openage', 'buildsystem'))
 
+    if args.cppstyle:
+        from .cppstyle import find_issues
+        yield from find_issues(check_files, ('openage', 'libopenage'))
```
Comment (MaanooAk) [inline buildsystem/codecompliance/__main__.py]: Ok (small change for next commit)

File: buildsystem/codecompliance/util.py [hunk 2]
```diff
@@ -49,6 +49,26 @@ def readfile(filename):
...
+            if filename.startswith('.'):
+                continue
+
+            filename = os.path.join(path, filename)
+
+            if os.path.isdir(filename):
+                yield from find_all_files((filename,))
+                continue
+
+            yield filename
```
Comment (TheJJ) [inline buildsystem/codecompliance/util.py]: Why did you copy the function? Just make a new case if the extension is None, then the extension isn't checked. then you can remove the duplicated code.

File: buildsystem/codecompliance/util.py [hunk 3]
```diff
@@ -68,3 +88,14 @@ def findfiles(paths, exts):
 
             if has_ext(filename, exts):
                 yield filename
+
+
+def issue_str(title, filename):
+    return (title, filename)
```
Comment (TheJJ) [inline buildsystem/codecompliance/util.py]: Can you factor out this function entirely? :)

**Comment by TheJJ** [comment]
I don't think there's a reason, we just used it to check all remaining files. So we can already use the infrastructure to check only `.cpp` and `.h` files efficiently with the extension check.

File: buildsystem/codecompliance/util.py (hunk 3 above)
Comment (MaanooAk) [inline buildsystem/codecompliance/util.py]: This function from which all (should/will) issues pass, so if we want to add other format or color we can just change this only function. You don't agree?

File: buildsystem/codecompliance/util.py (hunk 2 above)
Comment (MaanooAk) [inline buildsystem/codecompliance/util.py]: I thought I could make it smaller with out the extension checking (turns out I cant) and then forgot about it... I will change it

Comment (MaanooAk) [inline buildsystem/codecompliance/util.py]: I thought I could make it smaller with out the extension checking (turns out I cant) and then forgot about it... I will change it

File: buildsystem/codecompliance/util.py (hunk 3 above)
Comment (TheJJ) [inline buildsystem/codecompliance/util.py]: Ah, okay. Jup, then leave it and maybe add a comment what could be done there.

**Comment by MaanooAk** [comment]
This should be all done!

**Pull request merged**